hooks/pre-commit

```diff
@@ -0,0 +1,3 @@
+#! /usr/bin/env bash
+chmod +x
+git diff -U0 HEAD~2 HEAD | data/check-style --diff
```

Comment:
Should this definitely be HEAD~2 and not HEAD~1? I don't know much about pre-commit hooks, but if this is run before the commit, won't this be comparing the last 2 commits to the most recent and ignoring current changes? (I may well be wrong here, but would like to hear the explanation if so)

Reply:
Maybe git diff | data/check-style --diff will solve the problem given that we want to check the style of uncommitted changes

Reply:
I guess you'd need to diff the staging area, so I think `git diff --staged` rather than just `git diff` - I'm not sure if that's exactly right, but I'm sure some experimentation will reveal the answer :)

Reply:
I think` git diff --staged` works. I noticed that the style checking is however not consistent

Reply:
check-style gives different outputs with the same change when tried at different times

Reply:
Could you give an example of these different outputs vs the input and what you expect? Feel free to use https://gist.github.com/ if there's a lot of information you need to share.